docs/csharp/language-reference/unsafe-code.md

```diff
@@ -13,90 +13,106 @@ helpviewer_keywords:
-| [`fixed` statement](statements/fixed.md) | Temporarily fixes a variable so that its address can be found. |
+| `*`                                      | Dereference (pointer indirection).                             |
+| `->`                                     | Access struct member through a pointer.                        |
+| `[]`                                     | Index a pointer.                                               |
+| `&`                                      | Take the address of a variable.                                |
+| `++` and `--`                            | Increment/decrement pointer.                                   |
+| `+` and `-`                              | Pointer arithmetic.                                            |
+| `==`, `!=`, `<`, `>`, `<=`, and `>=`     | Pointer comparison.                                            |
+| [`stackalloc`](operators/stackalloc.md)  | Allocate memory on the stack.                                  |
+| [`fixed` statement](statements/fixed.md) | Pin a variable so its address can be taken.                    |
+
+See [Pointer-related operators](operators/pointer-related-operators.md) for details.
 
-For more information about pointer-related operators, see [Pointer-related operators](operators/pointer-related-operators.md).
+### Pointer conversions and Interop
 
-Any pointer type can be implicitly converted to a `void*` type. Any pointer type can be assigned the value `null`. Any pointer type can be explicitly converted to any other pointer type using a cast expression. You can also convert any integral type to a pointer type, or any pointer type to an integral type. These conversions require an explicit cast.
+- Any pointer type can be implicitly converted to `void*`.
+- Any pointer type can be set to `null`.
+- You can explicitly cast between pointer types and between pointers and integral types (integral types must be at least the size of a pointer: `nint`, `nuint`, `IntPtr`, `UIntPtr`, or — on 64-bit — `long`/`ulong`).
+- You can't dereference a `void*` directly, but you can cast it to another pointer type.
 
-The following example converts an `int*` to a `byte*`. Notice that the pointer points to the lowest addressed byte of the variable. When you successively increment the result, up to the size of `int` (4 bytes), you can display the remaining bytes of the variable.
+For example, converting an `int*` to a `byte*` lets you examine individual bytes:
 
 :::code language="csharp" source="snippets/unsafe-code/Conversions.cs" ID="Conversion":::
 
+### Pointer safety reminders
+
+- Dereferencing a null pointer is implementation-defined and may crash your program.
+- Passing pointers to or from methods, especially if they refer to stack or pinned data, can cause undefined behavior if the referent is no longer valid.
+- Never store a pointer to stack memory outside the current method.
```

Comment:
Perhaps it should be generalized like don't expose pointer to resources outside of their scopes? because it's definitely not stack-limited.